llvm/lib/Support/APFloat.cpp

```diff
@@ -136,6 +136,7 @@ static constexpr fltSemantics semIEEEquad = {16383, -16382, 113, 128};
 static constexpr fltSemantics semFloat8E5M2 = {15, -14, 3, 8};
 static constexpr fltSemantics semFloat8E5M2FNUZ = {
     15, -15, 3, 8, fltNonfiniteBehavior::NanOnly, fltNanEncoding::NegativeZero};
+static constexpr fltSemantics semFloat8E4M3 = {7, -6, 4, 8};
```

Comment:
Just sanity checking: this is 

```
maxExponent = 7
minExponent = -6
precision = 4
sizeInBits = 8
```
?

Reply:
Maksim, Adding f8E4M3 IEE754 dtype was split into two PRs:

1. llvm/clang/APFloat changes [PR-97179](https://github.com/llvm/llvm-project/pull/97179) 
2. MLIR changes [PR-97118](https://github.com/llvm/llvm-project/pull/97118)  (this PR)

PR-97118 depends on PR-97179. 

Currently you see two commits in PR-97118.

Once PR-97179 is merged - I'll rebase PR-97118 and it will contain just one commit - just MLIR changes.

Can you review just MLIR changes in this PR (PR-97118). All comments for llvm/clang/APFloat should go to [PR-97179](https://github.com/llvm/llvm-project/pull/97179).

Reply:
`f8E4M3` type follows IEEE 754 convention:
- Exponent bias: 7
- Maximum stored exponent value: 14 (binary 1110)
- Maximum unbiased exponent value: 14 - 7 = 7
- Minimum stored exponent value: 1 (binary 0001)
- Minimum unbiased exponent value: 1 − 7 = −6
- Precision specifies the total number of bits used for the significand (mantisa), including implicit leading integer bit = 3 + 1 = 4

`fltSemantics semFloat8E4M3 = {7, -6, 4, 8};`:
- maxExponent = 7
- minExponent = -6
- precision = 4
- sizeInBits = 8
- nonFiniteBehavior = fltNonfiniteBehavior::IEEE754 (default)
- nanEncoding = fltNanEncoding::IEEE (default)


The implementation patterns for `f8E4M3` should be similar to those of the existing `f8E5M2` type.